File: back/src/main/java/com/back/global/security/CustomUserDetailsService.java
package com.back.global.security;

import com.back.domain.member.entity.Member;
import com.back.domain.member.service.MemberService;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CustomUserDetailsService implements UserDetailsService {
    private final MemberService memberService;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        Member member = memberService.findByUsername(username);

        return new SecurityUser(
                member.getId(),
                member.getUsername(),
                member.getNickname(),
                member.getAuthorities()
        );
    }
}

File: back/src/main/kotlin/com/back/global/security/CustomUserDetailsService.kt
package com.back.global.security

import com.back.domain.member.entity.Member
import com.back.domain.member.service.MemberService
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class CustomUserDetailsService(
    private val memberService: MemberService
) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        val member: Member = memberService.getMemberByUsername(username)

        return SecurityUser(
            member.id,
            member.username!!,
            member.nickname,
            member.authorities
        )
    }
}